src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -605,16 +612,35 @@ public HttpResponse onSuccess(String authorizationCode) {
         };
     }
     
-    public void updateAuthorizationStrategy(OAuthSession oauth) throws IOException, GeneralSecurityException {
+    public UsernamePasswordAuthenticationToken updateAuthorizationStrategy(Credential credential) throws IOException, GeneralSecurityException {
     	populateDefaults();
-        OpenShiftUserInfo info = getOpenShiftUserInfo(oauth.getCredential(), transport);
-        Set<String> allowedRoles = postSAR(oauth.getCredential(), transport);
+        OpenShiftUserInfo info = getOpenShiftUserInfo(credential, transport);
+        Set<String> allowedRoles = postSAR(credential, transport);
         GrantedAuthority[] authorities = new GrantedAuthority[] {SecurityRealm.AUTHENTICATED_AUTHORITY};
+                
+        // we append the role suffix to the name stored into Jenkins, since a given user is able to log in at varying scope/permission
+        // levels in openshift; however, for now, we make sure the display name for Jenkins does not include this suffix
+        String suffix = "";
+        if (allowedRoles.contains("admin"))
+            suffix = "-admin";
+        else if (allowedRoles.contains("edit"))
+            suffix = "-edit";
+        else if (allowedRoles.contains("view"))
+            suffix = "view";
+        String matrixKey = info.getName() + suffix;
+        
         // logs this user in.
         UsernamePasswordAuthenticationToken token =
-                new UsernamePasswordAuthenticationToken(info.getName(), "", authorities);
+                new UsernamePasswordAuthenticationToken(matrixKey, "", authorities);
         SecurityContextHolder.getContext().setAuthentication(token);
         
+        User u = User.get(token.getName());
```

Comment:
It is the name of the account, if I gather what you mean by that correctly.  When I bring things up via `oc cluster up`, and log in as `developer`, `token.getName()` returns `developer`.  It is *not* the token contents.  For example, something that looks like `SQynup6xFEIshplTe-aj1mo-owqxx4n8SaiEgQvRa9E`.  And it is not the secret name.  For example, something that looks like `jenkins-token-m417e` (one of the secret names from my currently running jenkins service account).

I do not agree that multiple scoped tokens per user name is going to be a problem.  

Providing the token as input, the result of the self-sar requests  for the admin, edit, and view roles is what is important.  

Either whatever token provided has the admin, edit, or view role, or it does not.  Based on the role it has, the code maps it to one of the 3 correlating admin, edit or view permissions for the username our code will create in the Jenkins permission matrix.